using_images/other_images/jenkins.adoc

```diff
@@ -358,23 +358,41 @@ https://wiki.jenkins-ci.org/display/JENKINS/Kubernetes+Plugin[Kubernetes
 plug-in] that allows Jenkins slaves to be dynamically provisioned on multiple
 container hosts using Kubernetes and {product-title}.
 
-The Jenkins image entrypoint also provides auto-discovery and auto-configuration
-of the Kubernetes plug-ins by scanning the project Jenkins is deployed in for
+To facilitate the using of the Kubernetes plug-in, {product-title} provides three
+additional images for you consumption.
+
+The first is https://github.com/openshift/jenkins/tree/master/slave-base[base image] for Jenkins slaves.
+It pulls in both required tools (headless Java, the Jenkins JNLP client) and generally useful ones
+(including tar, zip, nss among others) as well as establishing the JNLP slave agent as the entrypoint.  And it provides
+Dockerfiles for both Centos and RHEL images.
+
+Two samples which extend this base image are provided:
+
+* https://github.com/openshift/jenkins/tree/master/slave-maven[Maven]
+* https://github.com/openshift/jenkins/tree/master/slave-nodejs[NodeJS]
+
+Both the Maven and NodeJS slave images are configured as Kubernetes Pod Tempate images within the {product-title} Jenkins image's
+configuration for the Kubernetes plugin.  That configuration includes labels for each which can
+be applied to any of your Jenkins jobs under their "Restrict where this project can be run" setting.  If the label is applied,
+execution of the given job will be done under an {product-title} Pod running the respective slave image.
+
+The two samples provide Dockerfiles for both Centos and RHEL that you can reference when building new slave images.
+Also note the `contrib` and `contrib/bin` subdirectories.  They allow for the insertion of configuration files and executable
+scripts for your image.
+
+The Jenkins image also provides auto-discovery and auto-configuration
+of slave images for the Kubernetes plug-ins by scanning the project Jenkins is deployed in for
 existing image streams with the label *role* set to *jenkins-slave*.
 
-When an image stream with this label is found, the entrypoint generates the
-corresponding Kubernetes plug-in configuration so you can assign your Jenkins
+When an image stream with this label is found, corresponding Kubernetes plug-in configuration
+is generated so you can assign your Jenkins
 jobs to run in a pod running the container image provided by the image stream.
```

Comment:
"Note: this scanning is only performed once, when the Jenkins master is starting.  If you label additional imagestreams, the Jenkins master will need to be restarted to pick up the additional images."